File: python/nwd1.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  nwd1.py


def main(args):
    a = int(input('Podaj liczbę a: '))
    b = int(input('Podaj liczbę b: '))
    licznik = 0
    while a != b:
        licznik += 1
        if a > b:
            a = a - b
        else:
            b = b - a
    print('NWD = ', a)
    print('Licznik: ', licznik)
    
    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: CPP/nwd1.cpp
/*
 * nwd1.cpp
 */


#include <iostream>
using namespace std;

int main(int argc, char **argv)
{
	int a, b;
    cout << "Podaj a: " << endl;
    cin >> a;
    cout << "Podaj b: " << endl;
    cin >> b;
    while (a != b){
        if(a > b){
            a = a - b;
            cout << a << endl;
        }
        else{
            b = b - a;
            cout << b << endl;
        }
    } 
    else if{
        cout << a << endl;
    }
	return 0;
}

